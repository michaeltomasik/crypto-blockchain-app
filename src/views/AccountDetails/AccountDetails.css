.AccountDetails-avatar, .AccountDetails-summary, .AccountDetails-radio, .AccountDetails-radio-label {
  margin-left: 15px;
  margin-right: 15px;
}

.AccountDetails-filters {
  display: flex;
  justify-content: flex-end;
  margin-bottom: 10px;
}

.AccountDetails-pagination {
  display: flex;
  justify-content: center;
}