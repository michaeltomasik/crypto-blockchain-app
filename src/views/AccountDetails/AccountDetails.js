import React from 'react';
import { Container } from 'react-bootstrap';

import Spinner from '../../components/Spinner';
import Summary from '../../components/Summary';
import TransactionList from '../../components/TransactionList';

import { TRANSACTION_FILTERS } from '../../constants/transactions';

class AccountDetails extends React.Component {
  constructor() {
    super();

    this.state = {
      filter: TRANSACTION_FILTERS.ALL.value,
      current: 1,
    }
  }

  componentDidMount() {
    this.loadAccount();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.address !== this.props.match.params.address) {
      this.loadAccount();
    }
  }

  loadAccount(data) {
    this.props.loadAccount({
      address: this.props.match.params.address,
      filter: this.state.filter,
      ...data,
    });
  }

  onRadioChange = (e) => {
    this.setState({
      filter: parseInt(e.target.value),
    }, () => {
      this.loadAccount();
    });
  }

  handlePageClick = page => {
    const perPage = 50;
    let offset = Math.ceil(page * perPage);

    this.loadAccount({ offset });
    this.setState({
      current: page,
    });
  };

  render() {
    const { address, final_balance, n_tx, txs } = this.props.account;

    if(this.props.isLoading) return <Spinner />
    if(!address) return 'No data';

    return (
      <div className="AccountDetails">
        <Container>
          <Summary data={{ address, final_balance, n_tx }} />
          { txs ?
            <TransactionList
              currentFilter={this.state.filter}
              onRadioChange={this.onRadioChange}
              txs={txs}
              n_tx={n_tx}
              currentPage={this.state.current}
              handlePageClick={this.handlePageClick}
              address={address} />
           : null}
        </Container>
      </div>
    );
  }
}

export default AccountDetails;
