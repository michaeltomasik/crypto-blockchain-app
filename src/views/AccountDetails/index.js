import { connect } from 'react-redux';
import AccountDetails from './AccountDetails';

const mapStateToProps = state => ({
  account: state.account,
});

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(AccountDetails);