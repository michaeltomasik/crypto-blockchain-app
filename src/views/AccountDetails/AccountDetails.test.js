import React, { PropTypes } from 'react';
import { Provider } from 'react-redux'
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';

import AccountDetails from './index';
import configureStore from 'redux-mock-store'
 
// create any initial state needed
const initialState = {
  account: {
    hash160: "8d5ead745127f0fdd0f8b4c3bf3987f90f75cc34",
    address: "3EaWcuNXQrW2JTGy5Cotm4pftovW8UfR2o",
    n_tx: 15808,
    total_received: 11851468851871,
    total_sent: 9660642776038,
    final_balance: 2190826075833,
    txs: [{
      ver: 2,
      weight: 756,
      inputs: [
        {
          prev_out: {
            spent: true,
            tx_index: 473608662,
            type: 0,
            addr: "12RqpZJ53TKTyNF6rm73YNvALqGbQH22d6",
            value: 54182667,
          }
        }
      ],
      out: [
        {
          spent: false,
          tx_index: 473611336,
          type: 0,
          addr: "3EaWcuNXQrW2JTGy5Cotm4pftovW8UfR2o",
          value: 54154317,
        }
      ],
      block_height: 587184,
      relayed_by: "0.0.0.0",
      lock_time: 0,
      result: 0,
      size: 189,
      block_index: 1775512,
      time: 1564176268,
    }]
  },
  loading: {},
};
const mockStore = configureStore();

let wrapper;
let store;

it('renders AccountDetails with data', () => {
  store = mockStore(initialState);
  wrapper = shallow(
    <MemoryRouter>
      <AccountDetails store={store}/>
    </MemoryRouter>
  );
  expect(wrapper.html()).toMatchSnapshot();
});

it('renders AccountDetails with no data', () => {
  store = mockStore({ account: {}, loading: {} });

  wrapper = shallow(
    <MemoryRouter>
      <AccountDetails store={store}/>
    </MemoryRouter>
  );
  expect(wrapper.html()).toMatchSnapshot();
});
