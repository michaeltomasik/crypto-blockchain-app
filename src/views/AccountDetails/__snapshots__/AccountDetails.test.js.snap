// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccountDetails with data 1`] = `"<div class=\\"AccountDetails\\"><div class=\\"container\\"><div class=\\"Summary\\"><h2>Summary</h2><div class=\\"row\\"><div class=\\"Summary-avatar\\"><canvas class=\\"identicon\\" style=\\"width:200px;height:200px\\"></canvas></div><div class=\\"Summary-info\\"><div><label class=\\"form-label\\"><b>Address</b></label><p>3EaWcuNXQrW2JTGy5Cotm4pftovW8UfR2o</p></div><div><label class=\\"form-label\\"><b>Balance</b></label><p>21908.26075833 BTC</p></div><div><label class=\\"form-label\\"><b>Transactions made</b></label><p>15808</p></div></div></div></div><div><h2>Transaction List</h2><div class=\\"justify-content-md-end row\\"><label class=\\"TransactionList-radio form-label\\"><b>Filters: </b></label><div class=\\"TransactionList-filters\\"><div class=\\"TransactionList-radio custom-control custom-radio\\"><input type=\\"radio\\" checked=\\"\\" value=\\"4\\" id=\\"All\\" class=\\"custom-control-input\\"/><label title=\\"\\" type=\\"checkbox\\" for=\\"All\\" class=\\"custom-control-label\\">All</label></div><div class=\\"TransactionList-radio custom-control custom-radio\\"><input type=\\"radio\\" value=\\"1\\" id=\\"Sent\\" class=\\"custom-control-input\\"/><label title=\\"\\" type=\\"checkbox\\" for=\\"Sent\\" class=\\"custom-control-label\\">Sent</label></div><div class=\\"TransactionList-radio custom-control custom-radio\\"><input type=\\"radio\\" value=\\"2\\" id=\\"Received\\" class=\\"custom-control-input\\"/><label title=\\"\\" type=\\"checkbox\\" for=\\"Received\\" class=\\"custom-control-label\\">Received</label></div></div></div><div class=\\"table-responsive-lg\\"><table class=\\"table TransactionTable\\"><thead><tr><th>Hash</th><th>Date</th><th>Amount</th></tr></thead><tbody><tr class=\\"row-transaction row-recieve \\"><td></td><td>7/26/2019, 11:24:28 PM</td><td>+ 0.54154317 BTC</td></tr><tr class=\\"TransactionTable-details\\"></tr></tbody></table></div><div class=\\"TransactionList-pagination\\"><ul class=\\"rc-pagination\\" unselectable=\\"unselectable\\"><li title=\\"上一页\\" class=\\"rc-pagination-disabled rc-pagination-prev\\" aria-disabled=\\"true\\"><a class=\\"rc-pagination-item-link\\"></a></li><li title=\\"1\\" class=\\"rc-pagination-item rc-pagination-item-1 rc-pagination-item-active\\" tabindex=\\"0\\"><a>1</a></li><li title=\\"2\\" class=\\"rc-pagination-item rc-pagination-item-2\\" tabindex=\\"0\\"><a>2</a></li><li title=\\"3\\" class=\\"rc-pagination-item rc-pagination-item-3\\" tabindex=\\"0\\"><a>3</a></li><li title=\\"4\\" class=\\"rc-pagination-item rc-pagination-item-4\\" tabindex=\\"0\\"><a>4</a></li><li title=\\"5\\" class=\\"rc-pagination-item rc-pagination-item-5 rc-pagination-item-before-jump-next\\" tabindex=\\"0\\"><a>5</a></li><li title=\\"向后 5 页\\" tabindex=\\"0\\" class=\\"rc-pagination-jump-next\\"><a class=\\"rc-pagination-item-link\\"></a></li><li title=\\"316\\" class=\\"rc-pagination-item rc-pagination-item-316\\" tabindex=\\"0\\"><a>316</a></li><li title=\\"下一页\\" tabindex=\\"0\\" class=\\" rc-pagination-next\\" aria-disabled=\\"false\\"><a class=\\"rc-pagination-item-link\\"></a></li></ul></div></div></div></div>"`;

exports[`renders AccountDetails with no data 1`] = `"No data"`;
